src/main/java/com/educare/unitylend/controller/UserController.java

```diff
@@ -30,9 +27,41 @@ public List<User> getAllUsers() throws ControllerException {
         try {
             List<User> userList = userService.getUsers();
             return userList;
-        } catch (ServiceException e) {
+        } catch (Exception e) {
             log.error("Error encountered in getting the users");
             throw new ControllerException("Error encountered in getting the users", e);
         }
     }
+
+    @PostMapping("/create-user")
+    public ResponseEntity<String> createUser(@RequestBody User user) throws ControllerException {
+        // Create the user
+        try {
+            userService.createUser(user);
+            return ResponseEntity.ok("succcesss!!!");
+        } catch (Exception e) {
+            log.error("Error encountered in getting the users");
+            throw new ControllerException("Error encountered in getting the users", e);
+        }
+
+    }
+
+    @PutMapping("/{userId}")
+    public ResponseEntity<String> updateUser(@PathVariable String userId, @RequestBody User updatedUser) throws ControllerException {
+        // Set the userId in the updatedUser object
+        updatedUser.setUserid(userId);
```

Comment:
Add a validation here if the passed `userId` is same as userId from the object `updatedUser`
Throw error in case of validation failure.

Reply:
Add null check for `userId`

Reply:
Yeah it has been done.